dmHelper/dmHelper.py

```diff
@@ -119,7 +119,9 @@ async def dmMember(self, ctx, member: discord.Member, *, message: str):
     @checks.admin_or_permissions(manage_guild=True)
     async def dmRole(self, ctx: commands.Context, role: discord.Role, *, message: str):
         """Sends a DM to all members with the specified role by adding them to the message queue"""
-        asyncio.create_task(self.add_message_players_to_dm_queue(members=role.members, content=message, ctx=ctx))
+        log.debug(f"Sending mass DM to {role.name}: {message}")
+        log.debug(f"{role.name} members: {[ x.name for x in role.members]}")
+        await self.add_message_players_to_dm_queue(members=role.members, content=message, ctx=ctx)
```

Comment:
I believe the `create_task` was intentional to allow this method to run in the background. I'm not sure what the actual tradeoffs here are between actually awaiting here and just letting it run in the background, but from my understanding it's definitely valid to have this here.

Reply:
Asyncio.create_task is used to make a function that is blocking "async". This is already async so there should be no need to create an individual thread for this task. I believe this is an artifact from an misunderstanding when I told Nully how sleep() was blocking in async code. However, this uses asyncio.sleep() which is completely safe.

Reply:
https://docs.python.org/3/library/asyncio-task.html#coroutine
I don't think that's quite right. Unless I've been reading this wrong, `create_task` is used to schedule the execution of a coroutine (such as an async method) in the background. If the desired behavior is to to schedule `add_message_players_to_dm_queue` in the background and move on to the next line, then this was correct.
The alternative would be to `await` like you have here, which is correct if we want to wait for this method to complete before moving onto the next line.
So what I'm saying is that both are valid, just depends on what we want, but in this instance it looks like it doesn't really matter too much lol.